WordPress/gutenberg
# [RNMobile] Gallery block: Add integration tests related to its manual tests

**Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/4532 and fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/3670.**





## Description


Include integration tests for the Gallery block based on the manual tests for that block ([reference](https://github.com/wordpress-mobile/test-cases/blob/trunk/test-cases/gutenberg/gallery.md)).

Apart from the new tests, the following additional changes will be incorporated:
- Media upload state constants are exposed in the `block-editor` package. This was required to test the flows of media management.
- The `testID` prop has been added to the media options picker, in order to be able to query it in the tests.
- A fix related to checking if a `Modal` component is visible has been applied to the custom matcher `toBeVisible`.
- After writing all the integration tests, the common parts have been extracted to a helpers file located within the Gallery block tests folder. In an upcoming PR, some of these helpers might be exposed globally.
**NOTE:** This change has required updating the Jest configuration to prevent helpers files to be considered test files.

## Testing Instructions




1. Run command `npm run native test` and observe that all tests succeed.
2. Observe that PR check `Unit Tests / Mobile` succeeds.

## Screenshots 
N/A

## Types of changes




Add integration tests

## Checklist:
